### PR TITLE
Feature/plus minus board

### PR DESCRIPTION
**Pull request by vatsalya25**
Changelogs:
- like functionality implemented
- edit functionality (currently disabled because it was using a text area which didn't look good)
- drag and drop enabled
- cards, box shadow, show delete and like on hover
- other UI changes

Issues:
- linear lists (two list items don't arrange themselves in one row)

### CODE REVIEW

File: src/app/action/action.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-<h2>Action Items!</h2>
+<h2 class="group-header">Action Items!</h2>
```
Comment (skeep) [inline src/app/action/action.html]: why is action items using same styling as sticky notes?

File: src/app/board/board.controller.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
 export class BoardController {
-  constructor ($firebaseArray, $location) {
+  constructor ($scope, $firebaseArray, $location) {
```
Comment (skeep) [inline src/app/board/board.controller.js]: why `$scope` injected?

File: src/app/board/board.controller.js [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
     'ngInject';
 
+    var userName = "";
```
Comment (skeep) [inline src/app/board/board.controller.js]: use `let`

File: src/app/board/board.controller.js [hunk 4]
```diff
@@ -13,9 +17,29 @@ export class BoardController {
...
+        $(this).removeClass("active");
+      });*/
+
+var anonymous = true;
+
+this.toggle = function() {
+  anonymous = !anonymous;
+  if(!anonymous){
+    userName = authData.google.displayName;
+    $('.anonymousToggle').css({"background-color":"#eeeeee","color":"black"});
```
Comment (skeep) [inline src/app/board/board.controller.js]: don't change styling from JS, use class

File: src/app/board/board.controller.js [hunk 5]
```diff
@@ -13,9 +17,29 @@ export class BoardController {
...
+  } else {
+    userName = "anonymous";
+    $('.anonymousToggle').css({"background-color":"#6D6A68","color":"white"});
```
Comment (skeep) [inline src/app/board/board.controller.js]: modifying DOM is againt angular principle. use directives like `ng-class`

File: src/app/board/board.controller.js [hunk 6]
```diff
@@ -26,7 +50,8 @@ export class BoardController {
           from: userName,
           uid: authData.uid,
           text: userMessage,
-          lane: id
+          lane: id,
+          dl: 0
```
Comment (skeep) [inline src/app/board/board.controller.js]: what's `dl` ... use descriptive name plz

File: src/app/board/board.controller.js [hunk 7]
```diff
@@ -26,7 +50,8 @@ export class BoardController {
           from: userName,
```
Comment (skeep) [inline src/app/board/board.controller.js]: define the message model first, possible in the constructor

File: src/app/board/board.controller.js [hunk 8]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
-        if (!event.shiftKey) $('#testForm').submit();
-    });*/
+    //
+    (new Firebase(roomURL)).on('child_added', (messagesObj) => {
+      (new Firebase(encodeURI(roomURL + "/" + messagesObj.key() + "/like"))).on('value', (userId) => {
+        if(userId.val() != null && messagesObj.key()) { //likes are there for this message
+          console.log(userId.numChildren());
+          console.log(messagesObj.val().uid);
+          //this.noOfLikes [messagesObj.val().uid] =userId.numChildren();
+          var fredNameRef = new Firebase(roomURL + "/" + messagesObj.key());
```
Comment (skeep) [inline src/app/board/board.controller.js]: do not use cryptic variable names

File: src/app/board/board.controller.js [hunk 9]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+        }
+      });
+});
 
-    //CODE TO MAKE THE USER ANONYMOUS
+    //handle like button click for a message
+    this.like = function(msg) {
+      //console.table("like: " + msg.$id);
+      //(new Firebase(roomURL)).child(msg.$id + "/like/" +authData.uid).set(1);
+      var t = (new Firebase(roomURL)).child(msg.$id + "/like/" +authData.uid);
```
Comment (skeep) [inline src/app/board/board.controller.js]: what's `t`

File: src/app/board/board.controller.js [hunk 10]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+        if(value.exists()){
+          t.remove();
+        }
+        else{
+          t.set(1);
+          t.off();
+        }
+      });
+
+      var ta = (new Firebase(roomURL)).child(msg.$id + "/like/");
```
Comment (skeep) [inline src/app/board/board.controller.js]: what's `ta`?

File: src/app/board/board.controller.js [hunk 11]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+      ta.once('value', function(snapshot) {
+        msg.noOfLikes=snapshot.numChildren();
+      });
+    }
+
+    //CODE TO ENABLE DRAG AND DROP OF STICKYs
+    $(function() {
```
Comment (skeep) [inline src/app/board/board.controller.js]: not angular way ... make custom directive of `sortable`

File: src/app/board/board.controller.js [hunk 12]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+      $('.delete-btn').css({'visibility' : 'hidden'});
+      $('.like-btn').css({'visibility' : 'hidden'});
+      $("#chat-messages-plus").sortable();
+      $("#chat-messages-minus").sortable();
+      $("#chat-messages-plus").disableSelection();
+      $("#chat-messages-minus").disableSelection();
+    });
+
+    //CODE TO DELETE THE MESSAGE POSTED
+    this.delete=function(msg){
```
Comment (skeep) [inline src/app/board/board.controller.js]: define these methods outside constructor.

File: src/app/board/board.controller.js [hunk 13]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
 
-    this.anonymous = function() {
-      userName = "anonymous";
+    //CODE TO SHOW DELETE/LIKE ETC ON HOVER
+    this.hover = function(msg){
+      var ide=msg.$id;
+      if(msg.uid === "google:"+authData.google.id) {
+
+      }
+      $('.delete-btn').css({'visibility' : 'visible'});
```
Comment (skeep) [inline src/app/board/board.controller.js]: not in JS, but CSS

File: src/app/board/board.controller.js [hunk 14]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+      // var roomRef = new Firebase('https://vyapi.firebaseio.com/messages/'+roomID).remove();
+      if(msg.uid === "google:"+authData.google.id)
+        this.msgRef.child(ide).remove();
+      console.log("delete: " + msg.$id);
+    };
 
-    this.anonymous = function() {
-      userName = "anonymous";
+    //CODE TO SHOW DELETE/LIKE ETC ON HOVER
+    this.hover = function(msg){
```
Comment (skeep) [inline src/app/board/board.controller.js]: why we need this in JS? why can't we use `:hover` in CSS ?

File: src/app/board/board.controller.js [hunk 15]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+    this.show = function(msg){
+      $('.delete-btn').css({'visibility' : 'hidden'});
+      $('.like-btn').css({'visibility' : 'hidden'});
+    };
+
+    //CODE TO EDIT THE STICKY NOTES
+    this.isOwner=function(msg)
+    {
+      var ide=msg.$id;
+      if(msg.uid=== "google:"+authData.google.id)
```
Comment (skeep) [inline src/app/board/board.controller.js]: simplify this `if` statment

File: src/app/board/board.controller.js [hunk 16]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+
+    this.show = function(msg){
+      $('.delete-btn').css({'visibility' : 'hidden'});
+      $('.like-btn').css({'visibility' : 'hidden'});
+    };
+
+    //CODE TO EDIT THE STICKY NOTES
+    this.isOwner=function(msg)
+    {
+      var ide=msg.$id;
```
Comment (skeep) [inline src/app/board/board.controller.js]: what's `ide`

File: src/app/board/board.controller.js [hunk 17]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+      if(msg.uid=== "google:"+authData.google.id)
+        return true;
+      else
+        return false;
+    }
+
+    this.currentMessageText= "hello";
```
Comment (skeep) [inline src/app/board/board.controller.js]: define all models and common variable in constructor first

File: src/app/board/board.controller.js [hunk 18]
```diff
@@ -37,27 +62,169 @@ export class BoardController {
...
+    this.currentMessage;
+    this.edit = function(msg) {
+      console.log("edit hererere");
+      this.currentMessageText= msg.text;
+      this.currentMessage = msg;
+    }
+    this.saveEdit = function (msg)
+    {
+      console.log( msg.text + "TT") ;
+      (new Firebase(roomURL + "/" + msg.$id+"/text")).set( msg.text);
```
Comment (skeep) [inline src/app/board/board.controller.js]: move this to a service

File: src/app/board/board.html [hunk 19]
```diff
@@ -4,13 +4,12 @@
...
-
-      <form id="testForm" ng-submit="board.submit('plus')">
+      <h2 class="group-header">What we did right!</h2>
+      <form ng-submit="board.submit('plus')">
         <div class="form-group">
           <div class="bg-success center-block" id="anonymousWarn" class="has-success anonymousWarn">Ssshh.. You're anonymous now</div>
           <div class="input-group">
-            <span type="button" id="anonymousToggle" class="input-group-addon btn btn-success anonymousToggle" ng-click="board.anonymous()">
+            <span type="button" id="anonymousTogglePlus" class="input-group-addon btn btn-success anonymousToggle" ng-click="board.toggle()">
               <span class="glyphicon glyphicon-eye-close pull-right"></span>
```
Comment (skeep) [inline src/app/board/board.html]: ise `<i>` for icon

File: src/app/board/board.scss [hunk 20]
```diff
@@ -1,6 +1,20 @@
 .userText {
   display: inline-block;
 }
+body {
+  background-color: #f4f1e8;
+  background-image: url("http://www.ashtangabutterfly.com/images/wood-grain-background-2.jpg");
```
Comment (skeep) [inline src/app/board/board.scss]: save image locally

File: src/app/board/board.scss [hunk 21]
```diff
@@ -1,6 +1,20 @@
 .userText {
   display: inline-block;
 }
+body {
+  background-color: #f4f1e8;
```
Comment (skeep) [inline src/app/board/board.scss]: use CSS shorthand

File: src/app/board/board.scss [hunk 22]
```diff
@@ -1,6 +1,20 @@
...
+  background-image: url("http://www.ashtangabutterfly.com/images/wood-grain-background-2.jpg");
+  /* This is mostly intended for prototyping; please download the pattern and re-host for production environments. Thank you! */
+}
+
+@font-face {
```
Comment (skeep) [inline src/app/board/board.scss]: define font-face in common file

File: src/app/board/board.scss [hunk 23]
```diff
@@ -1,6 +1,20 @@
...
+    font-family: myFirstFont;
+    src: url("https://fonts.googleapis.com/css?family=Average+Sans");
+}
+
+.group-header {
+  color: #E1E0DE;
+}
 
 #anonymousWarn {
```
Comment (skeep) [inline src/app/board/board.scss]: don't use ID to style

File: src/app/board/board.scss [hunk 24]
```diff
@@ -8,27 +22,87 @@
   padding: 10px;
 }
 
-.fixed-panel-head {
+#chat-messages-plus li { margin: 0 3px 3em 3px; padding: 0.4em; height: 220px; width: 180px}
```
Comment (skeep) [inline src/app/board/board.scss]: don't use ID to style

File: src/app/board/board.scss [hunk 25]
```diff
@@ -8,27 +22,87 @@
...
+#chat-messages-minus li { margin: 0 3px 3em 3px; padding: 0.4em; height: 220px; }
+/*#chat-messages-plus li span { position: absolute; margin-left: -1.3em; }*/
+
+.sticky {
+  float: left;
+  margin-left: 0.5em;
```
Comment (skeep) [inline src/app/board/board.scss]: use CSS shorthand

File: src/app/board/board.scss [hunk 26]
```diff
@@ -8,27 +22,87 @@
...
+  margin-bottom: 0.8em;
+  margin-right: 0.8em;
+  margin-top: 0.5em;
+  border-radius: 7px;
+  position: absolute;
```
Comment (skeep) [inline src/app/board/board.scss]: why is `position:absolute`

File: src/app/board/board.scss [hunk 27]
```diff
@@ -8,27 +22,87 @@
...
+  background-color: #FDFDFD;
+  background-image: url("http://www.transparenttextures.com/patterns/cream-pixels.png");
```
Comment (skeep) [inline src/app/board/board.scss]: save image locally

File: src/app/board/board.scss [hunk 28]
```diff
@@ -8,27 +22,87 @@
...
+
+.sticky {
+  float: left;
+  margin-left: 0.5em;
+  margin-bottom: 0.8em;
+  margin-right: 0.8em;
+  margin-top: 0.5em;
+  border-radius: 7px;
+  position: absolute;
+  background-color: #FDFDFD;
```
Comment (skeep) [inline src/app/board/board.scss]: use CSS shorthand

File: src/app/board/board.scss [hunk 29]
```diff
@@ -8,27 +22,87 @@
...
+  background-image: url("http://www.transparenttextures.com/patterns/cream-pixels.png");
+  /*border: 0.2em solid #6D6A68;*/
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  justify-content: space-around;
+  box-shadow: 8px 8px 20px;
```
Comment (skeep) [inline src/app/board/board.scss]: define color for box-shadow

File: src/app/board/board.scss [hunk 30]
```diff
@@ -8,27 +22,87 @@
...
+}
+
+.sticky__head {
   max-height: 30px;
   min-height: 30px;
   vertical-align: middle;
-  padding-bottom: 1px;
-  padding-top: 2px;
+  /*padding-bottom: 1px;*/
+  padding-top: 0.5em;
```
Comment (skeep) [inline src/app/board/board.scss]: CSS shrorthand plz

File: src/app/board/board.scss [hunk 31]
```diff
@@ -8,27 +22,87 @@
...
-  min-height: 60px;
+.sticky__body {
+  max-height: 190px;
+  min-height: 190px;
+  width: 100%;
+  height: 100%;
+  padding: 0.5em;
   overflow-y: auto;
   overflow-x: hidden;
   -ms-hyphens: auto;
```
Comment (skeep) [inline src/app/board/board.scss]: no need to add browser prefix, auto-prefixer task should do that for you

File: src/app/board/board.scss [hunk 32]
```diff
@@ -8,27 +22,87 @@
...
   -moz-hyphens: auto;
   -webkit-hyphens: auto;
   word-break: break-all;
+  color: black;
   hyphens: auto;
+  /*font-family: myFirstFont;*/
+}
+
+.sticky__body::-webkit-scrollbar {
```
Comment (skeep) [inline src/app/board/board.scss]: why you need this? it will only work in Safari or Chrome

File: src/app/board/board.scss [hunk 33]
```diff
@@ -8,27 +22,87 @@
...
 }
 
-.anonymousToggle {
-  background-color: #3c763d;
+.anonymous-toggle {
+  background-color: #6D6A68;
   color: white;
 }
+
+.glyphicon-remove-circle{
```
Comment (skeep) [inline src/app/board/board.scss]: do not override bootstrap class. why we need this?

File: src/app/board/board.scss [hunk 34]
```diff
@@ -8,27 +22,87 @@
...
+  border: none;
+  background-color: transparent;
+  color: #E34B2A;
+}
+
+.delete-btn:hover {
```
Comment (skeep) [inline src/app/board/board.scss]: do not iverride , bootstrap class, why we need it

File: src/app/board/board.scss [hunk 35]
```diff
@@ -8,27 +22,87 @@
...
+  display: inline;
+}
+
+.fa-thumbs-up {
```
Comment (skeep) [inline src/app/board/board.scss]: why we are overriding font-awesome class

File: src/app/room/room.css [hunk 36]
```diff
@@ -0,0 +1,20 @@
+body {
+  background-color: #F4F1E8;
+  background-image: url("http://www.ashtangabutterfly.com/images/wood-grain-background-2.jpg");
```
Comment (skeep) [inline src/app/room/room.css]: save image locally

File: src/app/room/room.css [hunk 37]
```diff
@@ -0,0 +1,20 @@
+body {
+  background-color: #F4F1E8;
```
Comment (skeep) [inline src/app/room/room.css]: use CSS shorthand

File: src/app/room/room.css [hunk 38]
```diff
@@ -0,0 +1,20 @@
...
+  background-image: url("http://www.ashtangabutterfly.com/images/wood-grain-background-2.jpg");
+  /* This is mostly intended for prototyping; please download the pattern and re-host for production environments. Thank you! */
+}
+
+
+.navbarItems {
```
Comment (skeep) [inline src/app/room/room.css]: do not use cameCase css class

File: src/app/room/room.css [hunk 39]
```diff
@@ -0,0 +1,20 @@
...
+  color: #6D6A68;
+}
+
+.navbar-header {
```
Comment (skeep) [inline src/app/room/room.css]: do not override bootstrap class

File: src/app/room/room.css [hunk 40]
```diff
@@ -0,0 +1,20 @@
...
+  color: #6D6A68;
+}
+
+.navbar {
```
Comment (skeep) [inline src/app/room/room.css]: do not override bootstrap class

File: src/app/room/room.html [hunk 41]
```diff
@@ -6,7 +6,7 @@
   <div class="container-fluid" ng-hide="getview">
 
   <div class="row">
-    <nav class = "navbar navbar-default" role = "navigation">
+    <nav class = "navbar navbar-default navbar--transparent" role = "navigation">
```
Comment (skeep) [inline src/app/room/room.html]: don;t think we need it. connect with @himanshusingh2407

File: src/app/user/user.scss [hunk 42]
```diff
@@ -1,34 +1,66 @@
...
-
-
 li.ng-leave.ng-leave-active,
 li.ng-enter {
-    opacity: 0;
-    -webkit-transition: opacity 300ms linear;
-    -moz-transition: opacity 300ms linear;
-    transition: opacity 300ms linear;
+  opacity: 0;
+  -webkit-transition: opacity 300ms linear;
```
Comment (skeep) [inline src/app/user/user.scss]: no need to use browser prefix

File: src/app/user/user.scss [hunk 43]
```diff
@@ -1,34 +1,66 @@
...
+  -moz-transition: opacity 300ms linear;
+  transition: opacity 300ms linear;
+}
+
+.user-image {
+  height: 4em;
+  padding-right: .25em;
```
Comment (skeep) [inline src/app/user/user.scss]: CSS shorthand plz

File: src/app/user/user.scss [hunk 44]
```diff
@@ -1,34 +1,66 @@
...
+
+.navbar-bottom {
+  text-align: center;
+  display: inline-block;
+  background-color: rgba(239,236,228, 0.5);
+}
+
+.user-image {
   height: 3em;
   padding-left: .25em;
```
Comment (skeep) [inline src/app/user/user.scss]: CSS shorthand plz

**Comment by skeep** [comment]
In general I would prefer smaller PR, as otherwise review becomes very difficult :)

File: src/app/action/action.html (hunk 1 above)
Comment (vatsalya25) [inline src/app/action/action.html]: Initially I was doing the styling for action part and it was supposed to look like board, that's why I used the same styling.. but now Abhimanyu is handling the action html and I have asked him to use different classes and different styling.

File: src/app/board/board.controller.js (hunk 2 above)
Comment (vatsalya25) [inline src/app/board/board.controller.js]: I wasn't working on the board controller actually. This probably had something to do with like or edit feature. Will ask and correct the code.

File: src/app/board/board.controller.js (hunk 4 above)
Comment (vatsalya25) [inline src/app/board/board.controller.js]: How to change the code in real time without using JS?

File: src/app/board/board.scss (hunk 34 above)
Comment (vatsalya25) [inline src/app/board/board.scss]: "delete-btn" is not a bootstrap class.. its a custom class that I've defined..

File: src/app/board/board.controller.js (hunk 4 above)
Comment (skeep) [inline src/app/board/board.controller.js]: use `ng-class`

File: src/app/action/action.html (hunk 1 above)
Comment (skeep) [inline src/app/action/action.html]: k

File: src/app/board/board.scss (hunk 34 above)
Comment (skeep) [inline src/app/board/board.scss]: ok, but have very similar names, its confusing. and why you need it?

File: src/app/board/board.controller.js (hunk 2 above)
Comment (rahuljain1311) [inline src/app/board/board.controller.js]: it is not needed will remove it in next PR

**Pull request merged**